examples/book-checkout.yaml

```diff
@@ -0,0 +1,57 @@
+
+'@context':
+  - https://git.io/vf-examples-jsonld-context
+  - library: https://library.example/
+    bob: https://bob.example/
+
+
+'@id': rgh:valueflows/valueflows/master/examples/book-checkout.yaml
+'@graph':
+
+  # resource before
+  - '@id': library:21f361a6-2375-46bb-b192-c21b5ba833bf
+    '@type': EconomicResource
+    skos:note: Valueflows cookbook
+    currentQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 1
+
+  - '@id': library:f47064eb-7120-4b31-b882-770165901fe6
+    '@type': Agreement
+    skos:note: renting Valueflows cookbook to Bob
+
+  - '@id': library:54b814ee-62dc-40c1-bb96-f8582aa4f771
+    '@type': EconomicEvent
+    clauseOf: library:f47064eb-7120-4b31-b882-770165901fe6
+    action: decrement
+    involves: library:21f361a6-2375-46bb-b192-c21b5ba833bf
+    involvesQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 1
+    time:inXSDDateTimeStamp: 2019-04-18T8:00:00-5:00
+    transfer:
+      - rights: use
+        provider: https://library.example/
+        receipient: https://bob.example/
+
+  - '@id': bob:98c2bfeb-9c70-4801-896c-4646b975a7d9
+    '@type': Commitment
+    clauseOf: library:f47064eb-7120-4b31-b882-770165901fe6
+    action: increment
+    involves: library:21f361a6-2375-46bb-b192-c21b5ba833bf
+    involvesQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 1
+    before:  2019-05-17T18:00:00-5:00
+    transfer:
+      - rights: use
+        provider: https://bob.example/
+        receipient: https://library.example/
+
+  # resource after
+  - '@id': library:21f361a6-2375-46bb-b192-c21b5ba833bf
+    '@type': EconomicResource
+    skos:note: Valueflows cookbook
+    currentQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 0
```

Comment:
Interestingly, I think the library wants to still know that it *owns* a certain number of the VF Cookbook, even if one or more is checked out and therefore not available.

Reply:
I tentatively think that the currentQuantity should refer to total "controlled", but maybe we need something else too - or maybe the availability can be calculated for now or later in the same way, by looking at commitments and events?

Reply:
They have commitment that this resource will get incremented again, they also should have track that they at some point in the past received *control* rights to that quantity of that resource and never transfer that right to anyone else so they still have that right.

Reply:
Yes, always an inventoried resource can be derived, it does not have to be instantiated at all.  But since we so far always instantiate it in practice, and have defined VF that way too, we need to define the currentQuantity carefully.  Do you think it should be the quantity you can see and touch always for that kind if resource, irrespective of if the agent actually "owns" all of it?